README.md

```diff
@@ -1,2 +1,107 @@
+- The `onChange` handlers on the other replicas will automatically detect the change in what node is the primary, and will execute `CHANGE MASTER` to change their replication source to the new primary.
+- At any point after this, we use `docker exec` to run `STOP SLAVE` on the new primary and can tear down the old primary if it still exists.
+
+---
+
+## Running the cluster
+
+Starting a new cluster is easy. Just run `docker-compose up -d` and in a few moments you'll have a running MySQL primary. Both the primary and replicas are described as a single `docker-compose` service. During startup, [Containerbuddy](http://containerbuddy.io) will ask Consul if an existing primary has been created. If not, the node will initialize as a new primary and all future nodes will self-configure replication with the primary in their `onStart` handler.
+
+Run `docker-compose scale mysql=2` to add a replica (or more than one!). Replication in this architecture uses [Global Transaction Idenitifers (GTID)](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids.html) rather than binlog positioning as this allows replicas to autoconfigure their position within the binlog. A primary that has the entire execution history can bootstrap a replica with no additional work. The replicas' `onChange` handler will automatically move replication to a new primary if one is created.
+
+So long as we're working with a new cluster that has never rotated the binlog on the primary (and where there's not too much data yet), the new replicas will be able to catch up to the primary without further intervention. But if you're bringing up a replica in an existing cluster, you'll need to try the following workflow.
+
+
+
+### Configuration
+
+Pass these variables in your environment or via an `_env` file.
+
+- `MYSQL_USER`: this user will be set up as the default non-root user on the node
+- `MYSQL_PASSWORD`: this user will be set up as the default non-root user on the node
+
+These variables are optional but you most likely want them:
+
+- `MYSQL_REPL_USER`: this user will be used on all instances to set up MySQL replication. If not set, then replication will not be set up on the replicas.
+- `MYSQL_REPL_PASSWORD`: this password will be used on all instances to set up MySQL replication. If not set, then replication will not be set up on the replicas.
+- `MYSQL_DATABASE`: create this database on startup if it doesn't already exist. The `MYSQL_USER` user will be granted superuser access to that DB.
+- `MANTA_URL`: the full Manta endpoint URL. (ex. `https://us-east.manta.joyent.com`)
+- `MANTA_USER`: the Manta account name.
+- `MANTA_SUBUSER`: the Manta subuser account name, if any.
+- `MANTA_ROLE`: the Manta role name, if any.
+- `MANTA_KEY_ID`: the MD5-format ssh key id for the Manta account/subuser (ex. `1a:b8:30:2e:57:ce:59:1d:16:f6:19:97:f2:60:2b:3d`).
+- `MANTA_PRIVATE_KEY`: the private ssh key for the Manta account/subuser.
```

Comment:
Here's my `_env` file:

```bash
# Environment variables for MySQL service
MYSQL_USER=someuser
MYSQL_PASSWORD= someuser
MYSQL_DATABASE= someuser
MYSQL_REPL_USER=anotheruser
MYSQL_REPL_PASSWORD= anotheruser

# Environment variables for backups to Manta
MANTA_BUCKET=/myuser/stor/triton-mysql
MANTA_USER= myuser
MANTA_KEY_ID=SHA256:some_sha
MANTA_URL=https://us-east.manta.joyent.com
```

It _doesn't_ have a `MANTA_PRIVATE_KEY`. Instead, I'm setting that in my shell environment with:

```bash
export MANTA_PRIVATE_KEY=`cat ~/.ssh/id_rsa`
```